lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -592,6 +593,9 @@ sub failed_modules_with_needles {
             for my $needle (@{$detail->{needles}}) {
                 push( @needles, [ $needle->{name}, $counter ] );
             }
+            if (!@{$detail->{needles}}) {
+                push(@needles, [ undef, $counter ] );
```

Comment:
Vladimir seems to be pretty consistent with {id} while you are consistent with {'id'} :)